public/pages/workflows/new_workflow/new_workflow.tsx

```diff
@@ -28,9 +28,68 @@ import {
 } from '../../../store';
 import { enrichPresetWorkflowWithUiMetadata } from './utils';
 import { getDataSourceId } from '../../../utils';
+import semver from 'semver';
+import { DataSourceAttributes } from '../../../../../../src/plugins/data_source/common/data_sources';
+import { getSavedObjectsClient } from '../../../../public/services';
 
 interface NewWorkflowProps {}
 
+export const getEffectiveVersion = async (
+  dataSourceId: string | undefined
+): Promise<string> => {
+  try {
+    // Don't make the API call if no dataSourceId
+    if (dataSourceId === undefined) {
+      console.log('cannot find data source');
+      throw new Error('Data source is required');
+    }
+
+    const dataSource = await getSavedObjectsClient().get<DataSourceAttributes>(
+      'data-source',
+      dataSourceId
+    );
+    const version = dataSource?.attributes?.dataSourceVersion || '2.17.0';
```

Comment:
I'm thinking, should we move '2.17.0' into common/constants as some default/constant var representing the minimum/default version? Can be helpful to add some comment as well highlighting what this is doing, something like, "If no datasource version is found, we default to 2.17.0" as it's not very intuitive behavior and logic we have to make here. Just some readability suggestions and may help in debugging future issues.

Reply:
addressed on line 55